src/nccl_ofi_rdma.c

```diff
@@ -5544,6 +5635,101 @@ static int release_ep(nccl_net_ofi_ep_t *base_ep)
 	return ret;
 }
 
+static nccl_net_ofi_rdma_ep_t *allocate_ep(nccl_net_ofi_rdma_device_t *device)
+{
+	int num_rails = device->num_rails;
+
+	/* Allocate endpoint */
+	nccl_net_ofi_rdma_ep_t *ep = calloc(1, sizeof(nccl_net_ofi_rdma_ep_t));
+	if (!ep) {
+		NCCL_OFI_WARN("Unable to allocate rdma endpoint");
+		return NULL;
+	}
+
+	/* Initialize base endpoint */
+	ep->base.device = &device->base;
+	ep->base.listen = listen;
+	ep->base.connect = connect;
+	ep->base.release_ep = release_ep;
+
+	/* Initialize number of rail */
+	ep->num_rails = num_rails;
+
+	/* Initialize reference count */
+	ep->ref_cnt = 0;
+
+	ep->bounce_buff_size = NCCL_OFI_MAX(NCCL_OFI_MAX(sizeof(nccl_net_ofi_rdma_ctrl_msg_t), eager_max_size),
+						sizeof(nccl_ofi_rdma_connection_info_t));
+
+	NCCL_OFI_TRACE(NCCL_NET, "RDMA endpoint %p for dev #%d is created",
+			ep,
+			device->base.dev_id);
+
+	return ep;
+}
+
+static int initialize_ep(nccl_net_ofi_rdma_ep_t *ep,
+			 nccl_net_ofi_rdma_device_t *device) {
+
+	int ret = 0;
+
+	ep->rails = calloc(ep->num_rails, sizeof(nccl_net_ofi_ep_rail_t));
+	if (!ep->rails) {
+		ret = -ENOMEM;
+		NCCL_OFI_WARN("Unable to allocate rdma rails");
+		return ret;
+	}
+
+	ret = nccl_ofi_deque_init(&ep->pending_reqs_queue);
+	if (ret != 0) {
+		NCCL_OFI_WARN("Failed to init pending_reqs_queue: %d", ret);
+		return ret;
+	}
+
+	ep->use_long_rkeys = device->use_long_rkeys;
+
+	ret = init_rail_ofi_resources(device, ep);
+	if (ret != 0) {
+		return ret;
+	}
+
+	ret = init_bounce_buffers(ep);
+	if (ret != 0) {
+		NCCL_OFI_WARN("Preparation of bounce buffers failed");
+		return ret;
+	}
+
+	/* Post all bounce buffers */
+	ret = post_bounce_buffs(ep);
+	if (ret != 0) {
+		NCCL_OFI_WARN("Posting of bounce buffers failed!");
+		return ret;
+	}
+
+	return ret;
+}
+
+static int create_new_ep(nccl_net_ofi_rdma_device_t *device,
```

Comment:
How is this different than allocate_ep and why have both?

Reply:
Consolidated into `create_ep`.